suite-native/video-assets/src/components/Video.tsx

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import { View, ActivityIndicator } from 'react-native';
+
+import { Video as ExpoVideo, VideoProps as ExpoVideoProps } from 'expo-av';
+
+import { useNativeStyles, prepareNativeStyle } from '@trezor/styles';
+
+import { videos, VideoName } from '../videos';
+
+type VideoProps = ExpoVideoProps & {
+    name: VideoName;
+    aspectRatio?: number;
+};
+
+type VideoStyleProps = {
+    aspectRatio: number;
+};
+
+const videoContainer = prepareNativeStyle((_, { aspectRatio }: VideoStyleProps) => ({
+    flex: 1,
+    aspectRatio,
+    justifyContent: 'center',
+    alignItems: 'center',
+}));
+
+const videoStyle = prepareNativeStyle((_, { aspectRatio }: VideoStyleProps) => ({
+    flex: 1,
+    aspectRatio,
+    resizeMode: 'contain',
+}));
+
+const activityIndicatorStyle = prepareNativeStyle(_ => ({
+    position: 'absolute',
+}));
+
+export const Video = ({
+    name,
+    shouldPlay = true,
+    isMuted = true,
+    aspectRatio = 1,
+    ...restProps
+}: VideoProps) => {
+    const [isLoading, setIsLoading] = useState(true);
+    const { applyStyle, utils } = useNativeStyles();
+    const videoSource = videos[name];
+
+    return (
+        <View style={applyStyle(videoContainer, { aspectRatio })}>
+            {isLoading && (
+                <ActivityIndicator
+                    size="large"
+                    color={utils.colors.green}
+                    style={applyStyle(activityIndicatorStyle)}
+                />
+            )}
+            <ExpoVideo
+                style={applyStyle(videoStyle, { aspectRatio })}
+                source={videoSource}
+                shouldPlay={shouldPlay}
+                isMuted={isMuted}
+                isLooping
+                onLoad={() => setIsLoading(false)}
```

Comment:
nit: use handler as you're doing everywhere else. Or don't do it everywhere else.

Reply:
Also I'm thinking that prop spreading should be as last, what do you think?

Reply:
fixed in 95199a91a